GLMakie/src/rendering.jl

```diff
@@ -125,7 +125,8 @@ function render_frame(screen::Screen; resize_buffers=true)
         # TODO, use actual boundingbox
         return Makie.zvalue2d(plot)
     end
-    sort!(screen.renderlist; by=sortby)
+    zvals = sortby.(screen.renderlist)
+    screen.renderlist = screen.renderlist[sortperm(zvals)]
```

Comment:
```suggestion
    permute!(screen.renderlist, sortperm(zvals))
```
To avoid some unnecessary allocations. 

Though I think it would be best to cache the z values instead (probably in plot?) and continue using `sort!`. I think @SimonDanisch wants to bring back cached bounding boxes - maybe caching z values is better done alongside that?

Reply:
Even if caching of z-values is on the roadmap like you say, could this be merged before that? It's quite nice as a render speed improvement